### ISSUE TITLE
Securitygroup.csv file layout

### ISSUE THREAD

**Issue by Begie1605** [microsoft/PartnerCenter-GDAPTransition#25]
To assign different roles to a id, how does the csv file be formated (specialy the column commaseperated roles).

We have this
Id,DisplayName,CommaSeperatedRoles
xxxxxxxxxxxxxxxx,HelpdeskAgents,"44367163-eba1-44c3-98af-f5787879f96a,a9ea8996-122f-4c74-9520-8edcd192826c,11648597-926c-4cf3-9c36-bcebb0ba8dcc,f2ef992c-3afb-46b9-b7cf-a126ee74c451,62e90394-69f5-4237-9190-012177145e10"

but this does not seem to be the correct layout?
can someone point me to the correct layout with the quotes and comma's?

**Comment by sansbacher** [comment]
That is correct. Any cell which has a comma _within it_ needs to be in **"double quotes"**. You can use the AAD RoleID or RoleName for the `CommaSeperatedRoles`. So this also works:

```
Id,DisplayName,CommaSeperatedRoles
806980ff-9f6c-43db-bba7-123456789,GDAP-Full Access,"Global Administrator, Global Reader"
3f6d5153-b6d0-45df-82d2-987654321,GDAP-Service Admins,"Global Reader, Compliance Administrator, Compliance Data Administrator, Conditional Access Administrator, Etc"
032df250-cc5b-4f9f-916c-112233445566,"GDAP-ReadOnly, ViewOnly, etc","f2ef992c-3afb-46b9-b7cf-a126ee74c451,4a5d8f65-41da-4de4-8968-e035b65339cf"
```

And the file needs to be: `GBM\GDAPBulkMigration\operations\securityGroup.csv`  
All of the AAD Roles must only be ones listed under `GDAP Migration\PartnerCenter-GDAPTransition\GBM\Configuration\ADRoleTemplatesAll.json`  

The `Id` and `DisplayName` columns are from your AAD Groups, which you should be downloading via the Tool to: `GBM\GDAPBulkMigration\downloads\securityGroup.csv`  
(Note: same file name, but a different file. The one in downloads is a list of ALL your AD Groups, the one in operations is the MAPPING from one or more of those groups to AAD Roles.

  --Saul

**Comment by Begie1605** [comment]
Hey Soul,
thnx for your comments,
I have verified the file customer .csv, adroles.csv and also securitygroup.csv in the operations directory and they seem to be in order.
The gdap relationship is created fine, but security groups are not.....and i have to setup these manually, what then agin works..

Any idea what still could be wrong...

These are the files in the operations folder...

adroles.csv
Id,Name,Description
44367163-eba1-44c3-98af-f5787879f96a,Dynamics 365 administrator,Can manage all aspects of the Dynamics 365 product.
a9ea8996-122f-4c74-9520-8edcd192826c,Power BI Administrator,Can manage all aspects of the Power BI product.
11648597-926c-4cf3-9c36-bcebb0ba8dcc,Power Platform Administrator,"Can create and manage all aspects of Microsoft Dynamics 365, PowerApps, and Microsoft Flows."
f2ef992c-3afb-46b9-b7cf-a126ee74c451,Global Reader,"Can read everything that a global administrator can, but not update anything."
62e90394-69f5-4237-9190-012177145e10,Groups Administrator,Can manage all aspects of Azure AD and Microsoft services that use Azure AD identities. Do not select this role unless absolutely necessary.


customer..csv
Name,PartnerTenantId,CustomerTenantId,OrganizationDisplayName,Duration
xxxxx,xxxxx,xxxx,xxxxx,730 (xxx is in real file also real values)

securitygroup.csv
Id,DisplayName,CommaSeperatedRoles
51452663-5dea-444c-ae7f-612c21dceadf,HelpdeskAgents,"44367163-eba1-44c3-98af-f5787879f96a,a9ea8996-122f-4c74-9520-8edcd192826c,11648597-926c-4cf3-9c36-bcebb0ba8dcc,f2ef992c-3afb-46b9-b7cf-a126ee74c451,62e90394-69f5-4237-9190-012177145e10"
bc2097ce-1317-45f4-916e-1eb91dfe2b6d,AdminAgents,"44367163-eba1-44c3-98af-f5787879f96a,a9ea8996-122f-4c74-9520-8edcd192826c,11648597-926c-4cf3-9c36-bcebb0ba8dcc,f2ef992c-3afb-46b9-b7cf-a126ee74c451,62e90394-69f5-4237-9190-012177145e10"

**Comment by sansbacher** [comment]
YOU need to create the Security Groups in YOUR tenant, then have the GDAP Tool dump out the `downloads\securityGroup.csv` (it will have ALL your Groups). Pick the Groups/IDs from that list.  

Then use them in the `operations\secuirtyGroup.csv` to set which Roles are assigned to each Group (the _CommaSeperatedRoles_ column). I would suggest NOT using the existing Groups, they may have special meaning, they were created by Microsoft and maybe MS will delete them after DAP is turned off?

Create new groups, like "GDAP-FullAdminAgents" and "GDAP-LimitedHelpdeskAgents" and assign your Roles to them. And make sure that **every time** you try running the Tool you are using a new _Name_ in the `operations\customers.csv` file (or you need to remove the existing GDAP relationship, but it just seems to be Terminated/Terminating, so best to use a new Name)

Also: if a particular step isn't working you'll have more luck if you can post any error messages the GDAP Tool is giving. Also look in `Gbm\Logs` to see if there's anything useful.

--Saul
